### PR TITLE
721 submit fail messages

### LINKED ISSUE #721: Submit failure messages (reports, step list, editing)
Currently, is something fails, users see a generic "Something went wrong" message. We'd like to provide provide users with more context if something unexpected happens.

- Report failed (red alert)
  - Header: "Something went wrong"
  - Body: 
       >      
       > We were unable to submit your report. Please try again. If you continue to experience problems, contact oec-data-pilot@skylight.digital for help.
- Save failed while editing (red alert)
  - Header: "Something went wrong"
  - Body: 
     > 
     > We were unable to save your changes. Please try again. If you continue to experience problems, contact oec-data-pilot@skylight.digital for help.

- Save failed while saving step list (red alert)
  - Header: "Something went wrong"
  - Body: 
     > 
     > We were unable to save the information you entered. Please try again. If you continue to experience problems, contact oec-data-pilot@skylight.digital for help.

### PR DESCRIPTION
**Pull request by mmazanec22**
Closes #721

### CODE REVIEW

File: src/Hedwig/ClientApp/src/containers/Enrollment/Detail/__snapshots__/EnrollmentDetail.test.tsx.snap [hunk 1]
```diff
@@ -30,7 +30,7 @@ exports[`EnrollmentDetail matches snapshot 1`] = `
         <span
           class="usa-hint text-italic"
         >
-          Last updated on February 12th, 2020 by T. User
+          Last updated on February 13th, 2020 by T. User
```
Comment (mmazanec22) [inline src/Hedwig/ClientApp/src/containers/Enrollment/Detail/__snapshots__/EnrollmentDetail.test.tsx.snap]: This seems like it's going to be annoying and maybe we should work around it somehow

Comment (jlhogan) [reply src/Hedwig/ClientApp/src/containers/Enrollment/Detail/__snapshots__/EnrollmentDetail.test.tsx.snap]: oh yeah! just update the `completeEnrollment` bootstrap data in `useApi` mock to have a fixed timestamp. my bad. I can push a commit if you want ! lmk

Comment (jlhogan) [reply src/Hedwig/ClientApp/src/containers/Enrollment/Detail/__snapshots__/EnrollmentDetail.test.tsx.snap]: cc @mmazanec22  ^

Comment (jlhogan) [reply src/Hedwig/ClientApp/src/containers/Enrollment/Detail/__snapshots__/EnrollmentDetail.test.tsx.snap]: fixing this on https://github.com/ctoec/ecis-experimental/pull/756

Comment (cailyncodes) [reply src/Hedwig/ClientApp/src/containers/Enrollment/Detail/__snapshots__/EnrollmentDetail.test.tsx.snap]: Fix in master

**Pull request merged**